istioctl/cmd/revision.go

```diff
@@ -205,9 +205,9 @@ func revisionList(writer io.Writer, args *revisionArgs, logger clog.Logger) erro
 		if ri := revisions[rev]; ri == nil {
 			revisions[rev] = &tag.RevisionDescription{}
 		}
-		cs, err := getEnabledComponents(iop)
+		cs, err := getEnabledUserFacingComponents(iop)
 		if err != nil {
-			return fmt.Errorf("error while getting IstioOperator Components: %v", err)
+			logger.LogAndErrorf("error while getting IstioOperator %s/%s Components: %v", iop.Namespace, iop.Name, err)
```

Comment:
Why changed the error return logic here? If there's an error, looks like the IOP is invalid, which should return error instead of logging and skipping?

Reply:
I think that if only the parsing component is not working properly, it should not be considered a blocking error. A warning message is enough in this case.